Sources/MacroToolkit/NormalizedTypes/NormalizedCompositionType.swift

```diff
@@ -0,0 +1,12 @@
+import SwiftSyntax
+
+/// Wraps a composition type (e.g. `ProtocolA & ProtocolB`).
+public struct NormalizedCompositionType: TypeProtocol {
```

Comment:
Add a way to access the types that have been composed. Even though they don't need to be recursively normalized, I think it'd still be good to represent them as `NormalizedSimpleType`s so that they can be used as if they've been normalized.

Reply:
I based the implementation of `NormalizedCompositionType` on `CompositionType`, that has no direct way to access the sub-types. Should I still implement this property on `NormalizedCompositionType`?